# 1ª Tarefa

É possível acessar SIM os arquivos compactos diretamente do site sem precisar baixá-los.

In [25]:
import requests
import pandas as pd
from zipfile import ZipFile
from io import BytesIO

# Agrupados por pessoa - Todas as causas e tipos de acidentes (a partir de 2018)


url2020 = "https://arquivos.prf.gov.br/arquivos/index.php/s/hXimwPNR9lyqdMS/download"
url2019 = "https://arquivos.prf.gov.br/arquivos/index.php/s/sdvJndbl5wLyh3J/download"
url2018 = "https://arquivos.prf.gov.br/arquivos/index.php/s/EF4uPKCihT0ouXd/download"

In [4]:
acidentes2020 = pd.read_csv(url2020, compression='zip',encoding='iso8859-1', sep=';')
acidentes2020.head()

id     pesid data_inversa dia_semana   horario  uf     br     km  \
0  260031.0  578987.0   2020-01-01     quarta  01:00:00  TO  153.0  678,1   
1  260031.0  578986.0   2020-01-01     quarta  01:00:00  TO  153.0  678,1   
2  260031.0  578988.0   2020-01-01     quarta  01:00:00  TO  153.0  678,1   
3  260031.0  578991.0   2020-01-01     quarta  01:00:00  TO  153.0  678,1   
4  260031.0  578475.0   2020-01-01     quarta  01:00:00  TO  153.0  678,1   

  municipio causa_principal  ...       sexo  ilesos feridos_leves  \
0    GURUPI             Sim  ...  Masculino       0             0   
1    GURUPI             Sim  ...   Feminino       0             0   
2    GURUPI             Sim  ...  Masculino       0             0   
3    GURUPI             Sim  ...   Feminino       0             0   
4    GURUPI             Sim  ...   Feminino       0             0   

  feridos_graves mortos      latitude     longitude regional delegacia  uop  
0              1      0  -11,77460203  -49,10744996    SR-TO  UOP01/TO  NaN  
1              1      0  -11,77460203  -49,10744996    SR-TO  UOP01/TO  NaN  
2              1      0  -11,77460203  -49,10744996    SR-TO  UOP01/TO  NaN  
3              1      0  -11,77460203  -49,10744996    SR-TO  UOP01/TO  NaN  
4              1      0  -11,77460203  -49,10744996    SR-TO  UOP01/TO  NaN  

[5 rows x 37 columns]

In [5]:
acidentes2019 = pd.read_csv(url2019, compression='zip',encoding='iso8859-1', sep=';')
acidentes2019.head()       

id     pesid data_inversa   dia_semana   horario  uf     br     km  \
0  182210.0  402103.0   2019-01-01  terça-feira  01:30:00  SP  116.0    218   
1  182210.0  402106.0   2019-01-01  terça-feira  01:30:00  SP  116.0    218   
2  182210.0  402104.0   2019-01-01  terça-feira  01:30:00  SP  116.0    218   
3  182210.0  402102.0   2019-01-01  terça-feira  01:30:00  SP  116.0    218   
4  182211.0  402126.0   2019-01-01  terça-feira  01:30:00  PR  373.0  177,3   

      municipio causa_principal  ...       sexo  ilesos feridos_leves  \
0     GUARULHOS             Sim  ...  Masculino       0             1   
1     GUARULHOS             Sim  ...  Masculino       0             1   
2     GUARULHOS             Sim  ...   Feminino       0             1   
3     GUARULHOS             Sim  ...  Masculino       0             1   
4  PONTA GROSSA             Sim  ...  Masculino       0             1   

  feridos_graves mortos      latitude     longitude regional delegacia  \
0              0      0  -23,46052014  -46,48772478    SR-SP    DEL6/1   
1              0      0  -23,46052014  -46,48772478    SR-SP    DEL6/1   
2              0      0  -23,46052014  -46,48772478    SR-SP    DEL6/1   
3              0      0  -23,46052014  -46,48772478    SR-SP    DEL6/1   
4              0      0  -25,05533957  -50,22776753    SR-PR    DEL7/3   

        uop  
0  UOP01/SP  
1  UOP01/SP  
2  UOP01/SP  
3  UOP01/SP  
4  UOP01/PR  

[5 rows x 37 columns]

In [6]:
acidentes2018 = pd.read_csv(url2018, compression='zip',encoding='iso8859-1', sep=';')
acidentes2018.head()

id     pesid data_inversa     dia_semana   horario  uf     br     km  \
0  99973.0  216523.0   2018-01-01  segunda-feira  00:20:00  RJ  116.0  303,5   
1  99973.0  216524.0   2018-01-01  segunda-feira  00:20:00  RJ  116.0  303,5   
2  99973.0  216532.0   2018-01-01  segunda-feira  00:20:00  RJ  116.0  303,5   
3  99973.0  216527.0   2018-01-01  segunda-feira  00:20:00  RJ  116.0  303,5   
4  99973.0  216530.0   2018-01-01  segunda-feira  00:20:00  RJ  116.0  303,5   

  municipio causa_principal  ...       sexo  ilesos feridos_leves  \
0   RESENDE             Sim  ...  Masculino       0             0   
1   RESENDE             Sim  ...   Feminino       0             0   
2   RESENDE             Sim  ...  Masculino       1             0   
3   RESENDE             Sim  ...   Feminino       0             0   
4   RESENDE             Sim  ...  Masculino       0             0   

  feridos_graves mortos   latitude  longitude regional delegacia       uop  
0              1      0  -22,46937  -44,44705    SR-RJ    DEL5/7  UOP03/RJ  
1              1      0  -22,46937  -44,44705    SR-RJ    DEL5/7  UOP03/RJ  
2              0      0  -22,46937  -44,44705    SR-RJ    DEL5/7  UOP03/RJ  
3              1      0  -22,46937  -44,44705    SR-RJ    DEL5/7  UOP03/RJ  
4              1      0  -22,46937  -44,44705    SR-RJ    DEL5/7  UOP03/RJ  

[5 rows x 37 columns]

# 2ª Tarefa

(a) quantos acidentes ocorreram em cada ano?

In [8]:
len(acidentes2020)

394482

Em 2020 ocorreram 394482 acidentes

In [9]:
len(acidentes2019)

331666

Em 2019 ocorreram 331666 acidentes

In [10]:
len(acidentes2018)

324809

Em 2018 ocorreram 331666 acidentes

(b) quantas variáveis cada conjunto de dados registra?

In [13]:
acidentes2020.shape

(394482, 37)

In [16]:
acidentes2019.shape

(331666, 37)

In [17]:
acidentes2018.shape

(324809, 37)

In [28]:
acidentes2018.columns.tolist()

['id',
 'pesid',
 'data_inversa',
 'dia_semana',
 'horario',
 'uf',
 'br',
 'km',
 'municipio',
 'causa_principal',
 'causa_acidente',
 'ordem_tipo_acidente',
 'tipo_acidente',
 'classificacao_acidente',
 'fase_dia',
 'sentido_via',
 'condicao_metereologica',
 'tipo_pista',
 'tracado_via',
 'uso_solo',
 'id_veiculo',
 'tipo_veiculo',
 'marca',
 'ano_fabricacao_veiculo',
 'tipo_envolvido',
 'estado_fisico',
 'idade',
 'sexo',
 'ilesos',
 'feridos_leves',
 'feridos_graves',
 'mortos',
 'latitude',
 'longitude',
 'regional',
 'delegacia',
 'uop']

Nos dataframes correspondentes aos anos de 2020, 2019 e 2018 existem, em cada um, 37 variáveis.

(c) quais as cinco cidades brasileiras onde mais ocorreram acidentes em rodovias federais?

In [30]:
acidentes2020['municipio'].mode() # em 2020 Brasília foi a cidade com maior registro de acidentes

0    BRASILIA
dtype: object

In [33]:
acidentes2019['municipio'].mode() # em 2019 Brasília foi a cidade com maior registro de acidentes

0    BRASILIA
dtype: object

In [35]:
acidentes2018['municipio'].mode() # em 2018 Curitiba foi a cidade com maior registro de acidentes

0    CURITIBA
dtype: object

(d) quantos acidentes com feridos graves aconteceram na Paraíba em 2019?

In [20]:
acidentes2019['estado_fisico'][(acidentes2019['estado_fisico'] == 'Lesões Graves') & (acidentes2019['uf'] == 'PB')].count()

1123

No estado da Paraíba ocorreram 1123 acidentes com lesões graves no ano de 2019

# Tarefa 3    


In [21]:
trienioPorEstado = pd.concat([acidentes2020[['id', 'uf']], acidentes2019[['id', 'uf']], acidentes2018[['id', 'uf']]], ignore_index = True)
trienioPorEstado.mode()

id  uf
0  120498.0  MG

(b) acidentes por dia da semana para todos os anos (um só arquivo combinando a informação dos três anos).


In [22]:
trienioPorDia = pd.concat([acidentes2020[['id', 'dia_semana']], acidentes2019[['id', 'dia_semana']], acidentes2018[['id', 'dia_semana']]], ignore_index = True)
trienioPorDia

id     dia_semana
0        260031.0         quarta
1        260031.0         quarta
2        260031.0         quarta
3        260031.0         quarta
4        260031.0         quarta
...           ...            ...
1050952  253040.0    sexta-feira
1050953  253040.0    sexta-feira
1050954  253040.0    sexta-feira
1050955  253040.0    sexta-feira
1050956  253078.0  segunda-feira

[1050957 rows x 2 columns]

# Tarefa 4

(a) qual a causa mais frequente e a mais rara de acidentes registrados nos três anos?

In [24]:
trienio = pd.concat([acidentes2020, acidentes2019, acidentes2018], ignore_index = True)
trienio['causa_acidente'].mode() #causa mais frequente de acidentes

0    Falta de Atenção à Condução
dtype: object